File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minimum = INT_MAX;
        int maximum = 0;
    
        for (int i = 0; i < prices.size(); i++) {
        if (prices[i] < minimum) {
            minimum = prices[i];
        } else if (prices[i] - minimum > maximum) {
            maximum = prices[i] - minimum;
        }
        }
    
        return maximum;
    
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
      minimum = float("inf")
      max_profit = 0

      for i in prices:
        if i < minimum:
          minimum = i
        elif i - minimum > max_profit:
          max_profit = i - minimum
      
      return max_profit
        

"""

minimum = Infinity
maxProfit = -Infinity

iterate through prices
  if price < minimum: set minimum to prices
  if price - minimum > maxProfit: set maxProfit to difference

return maxProfit

"""